src/test/perl/path.t

```diff
@@ -277,63 +298,242 @@ $mc->{fail} = undef;
 is($mc->_untaint_path("abc", "ok"), "abc", "untaint ok");
 ok(! defined($mc->{fail}), "no fail attribute set with ok path");
 
-=head2 directory/file/any exists
+
+=head2 directory/file/any exists on files and directoriees
 
 =cut
 
-init_exception("existence tests");
+init_exception("existence tests (file/directory)");
 
 # Tests without NoAction
 $CAF::Object::NoAction = 0;
 
-rmtree if -d $basetest;
+rmtree ($basetest) if -d $basetest;
 ok(! $mc->directory_exists($basetest), "directory_exists false on missing directory");
 ok(! $mc->file_exists($basetest), "file_exists false on missing directory");
 ok(! $mc->any_exists($basetest), "any_exists false on missing directory");
+ok(! $mc->is_symlink($basetest), "is_symlink false on missing directory");
 
 ok(! $mc->directory_exists($basetestfile), "directory_exists false on missing file");
 ok(! $mc->file_exists($basetestfile), "file_exists false on missing file");
 ok(! $mc->any_exists($basetestfile), "any_exists false on missing file");
+ok(! $mc->is_symlink($basetestfile), "is_symlink false on missing file");
 
 makefile($basetestfile);
 
 ok($mc->directory_exists($basetest), "directory_exists true on created directory");
 ok($mc->any_exists($basetest), "any_exists true on created directory");
 ok(! $mc->file_exists($basetest), "file_exists false on created directory");
+ok(! $mc->is_symlink($basetest), "is_symlink false on created directory");
 
 ok(! $mc->directory_exists($basetestfile), "directory_exists false on created file");
 ok($mc->any_exists($basetestfile), "any_exists true on created file");
 ok($mc->file_exists($basetestfile), "file_exists true on created file");
+ok(! $mc->is_symlink($basetestfile), "is_symlink false on created file");
+
+# noreset=1
+verify_exception("existence tests (file/directory)", undef, 0, 1);
+ok($mc->_reset_exception_fail(), "_reset_exception_fail after existence tests (file/directory)");
+
+
+=head2 symlink/hardlink creation/update/test
+
+=cut
+
+# Test symlink creation
+
+# Function to do the ok()/is() pair, taking into account NoAction flag
+sub check_symlink {
+    my ($mc, $target, $link_path, $expected_status) = @_;
+    my $noaction = $mc->_get_noaction();
```

Comment:
since you do not expect/support keeps_state, might as well use `CAF::Object::NoAction`